### PR TITLE
feat(__main__.py): Add support for width, height, and scale query  strings

### PR DESCRIPTION
**Pull request by laithalsaadoon**
Thanks for a great and helpful library!

### CODE REVIEW

**Comment by laithalsaadoon** [comment]
Closes #168

File: mermaid/__main__.py [hunk 1]
```diff
@@ -43,12 +44,36 @@ def __init__(
 
         Parameters:
             graph (Graph): The Graph object containing the Mermaid diagram script.
+            width (Optional[int]): The width of the SVG image.
+            height (Optional[int]): The height of the SVG image.
+            scale (Optional[int]): The scale of the SVG image.
+            position (Union[Position, str]): The position of the node in the Mermaid diagram.
         """
```
Comment (ouhammmourachid) [inline mermaid/__main__.py:51]: here you can add assertion to validate the scale
```python
# Validate the scale parameter if it is provided and validate the width and height parameters if scale is provided
        if scale:
            assert 1 <= scale <= 3, "Scale must be between 1 and 3"
            assert any([width, height]), "One or both of width and height must be provided"
 ```
 and also  in the docs  of mermaid.ink the `scale` can be float between 1 and 3

File: mermaid/__main__.py [hunk 2]
```diff
@@ -43,12 +44,36 @@ def __init__(
...
         self.__position: str = position if isinstance(position, str) else position.value
+        self.__height = height if height else None
+        self.__width = width if width else None
+        self.__scale = scale if scale else None
+
         self._diagram = self._process_diagram(graph.script)
-        self.width, self.height, self.scale = width, height, scale
+
+        if any([self.__width, self.__height, self.__scale]):
+            self._diagram += "&" + self._build_query_params()
```
Comment (ouhammmourachid) [inline mermaid/__main__.py:60]: replace `&` by `?`

File: tests/test_mermaid.py
Comment (ouhammmourachid) [inline tests/test_mermaid.py:1]: here  add some cases in witch the creation of mermaid class should raise `AssertionError` related to `scale`

````python
def test_should_raise_error_on_invalid_scale(self):
        with self.assertRaises(AssertionError):
            Mermaid(self.graph, scale=2)

        with self.assertRaises(AssertionError):
            Mermaid(self.graph, height=100, scale=4)

        with self.assertRaises(AssertionError):
            Mermaid(self.graph, width=100, scale=0.5)

**Review by ouhammmourachid — CHANGES_REQUESTED** [review]

File: tests/test_mermaid.py
Comment (laithalsaadoon) [reply tests/test_mermaid.py:1]: Nice, good idea! Added

File: mermaid/__main__.py (hunk 2 above)
Comment (laithalsaadoon) [reply mermaid/__main__.py]: Woops, good catch :)

File: mermaid/__main__.py (hunk 1 above)
Comment (laithalsaadoon) [reply mermaid/__main__.py:52]: Added, good catch

File: mermaid/__main__.py [hunk 3]
```diff
@@ -84,6 +88,17 @@ def _make_request_to_mermaid(self) -> None:
...
         the base64 encoded string of the Mermaid diagram script.
         """
+        # update self._diagram with optional query strings based on self.width, self.height, self.scale. All three are optional and can be used all at once, so the query string needs to be valid
+
+        query_string = ''
+        for param, value in [('width', self.width), ('height', self.height), ('scale', self.scale)]:
+            if value is not None:
+                query_string += f'&{param}={value}'
+        if query_string.startswith('&'):
+            query_string = '?' + query_string[1:]
```
[the remark this answers is not in the document]
Comment (laithalsaadoon) [reply mermaid/__main__.py]: Thanks @ouhammmourachid ! Ironically I moved to it's own function with urllib separate it out. I addressed all your comments

**Review by ouhammmourachid — APPROVED** [review]
Approved! Thanks for the update.

**Pull request merged**